File: Homework_2/app/src/main/java/com/example/homework_2/models/U2.java
package com.example.homework_2.models;

public class U2 extends Rocket{
    public U2 (){
        setCost(120);
        setWeight(18000);
        setMaxWeight(29000);
        setCurrentWeight(getWeight());
    }

    public boolean launch() {
        int random = (int)(Math.random() * 100 + 1);
        return 4.0 * (getCurrentWeight() - getWeight()) / (getMaxWeight() - getWeight()) <= random;
    }

    public boolean land() {
        int random = (int)(Math.random() * 100 + 1);
        return 8.0 * (getCurrentWeight() - getWeight()) / (getMaxWeight() - getWeight()) <= random;
    }
}

File: SpaceShip_Kotlin/app/src/main/java/com/example/spaceship_kotlin/moduls/U2.kt
package com.example.spaceship_kotlin.moduls

class U2: Rocket() {
    init {
        cost = 120
        weight = 18000
        maxWeight = 29000
        currentWeight = weight
    }

    override fun land(): Boolean {
        val random = (Math.random() * 100 + 1).toInt()
        return 8.0 * (currentWeight - weight) / (maxWeight - weight) <= random
    }

    override fun launch(): Boolean {
        val random = (Math.random() * 100 + 1).toInt()
        return 4.0 * (currentWeight - weight) / (maxWeight - weight) <= random
    }
}